superset/assets/src/dashboard/reducers/getInitialState.js

```diff
@@ -98,20 +102,22 @@ export default function(bootstrapData) {
 
       sliceIds.add(key);
 
-      // if chart is newly added from explore view, add a row in layout
+      // if there are newly added slices from explore view, add 3 slices-per-row into layout
       if (!chartIdToLayoutId[key] && layout[parentId]) {
-        const parent = layout[parentId];
-        const rowContainer = newComponentFactory(ROW_TYPE);
-        layout[rowContainer.id] = rowContainer;
-        parent.children.push(rowContainer.id);
+        if (newSlicesCounter % 3 === 0) {
```

Comment:
I think a more robust way to do this is to use the values from `constants.js` and the actual width of the newly-created `chartHolder` component. If the current row doesn't have enough width for the new component, then you can add a new row.

Your approach would break if someone unknowingly changed the default chart width.

Reply:
fixed per comment.